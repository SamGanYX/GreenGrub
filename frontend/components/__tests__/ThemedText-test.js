import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from '../Themed'; // Adjust path if needed

describe('Themed.Text', () => {
  it('renders correctly with default theme colors', () => {
    const tree = renderer
      .create(<Text>This is themed text</Text>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('applies custom light and dark colors when provided', () => {
    const tree = renderer
      .create(
        <Text lightColor="#000000" darkColor="#ffffff">
          Custom colored text
        </Text>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});