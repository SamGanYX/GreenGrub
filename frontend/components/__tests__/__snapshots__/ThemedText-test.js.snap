// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themed.Text applies custom light and dark colors when provided 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
      undefined,
    ]
  }
>
  Custom colored text
</Text>
`;

exports[`Themed.Text renders correctly with default theme colors 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  This is themed text
</Text>
`;
